src/components/stepper/step-label/step-label.ts

```diff
@@ -0,0 +1,88 @@
+import { type CSSResultGroup, html, type TemplateResult } from 'lit';
+import { customElement } from 'lit/decorators.js';
+
+import {
+  SbbButtonBaseElement,
+  SbbDisabledTabIndexActionMixin,
+  SbbIconNameMixin,
+  hostAttributes,
+} from '../../core/common-behaviors';
+import { ConnectedAbortController } from '../../core/eventing';
+import type { SbbStepElement } from '../step/step';
+import type { SbbStepperElement } from '../stepper';
+
+import style from './step-label.scss?lit&inline';
+import '../../icon';
+
+let nextId = 0;
+
+/**
+ * Combined with a `sbb-stepper`, it displays a step's label.
+ *
+ * @slot - Use the unnamed slot to provide a label.
+ */
+@customElement('sbb-step-label')
+@hostAttributes({
+  slot: 'step-label',
+  role: 'tab',
+})
+export class SbbStepLabelElement extends SbbIconNameMixin(
+  SbbDisabledTabIndexActionMixin(SbbButtonBaseElement),
+) {
+  public static override styles: CSSResultGroup = style;
+
+  /** @internal */
+  public readonly internals: ElementInternals = this.attachInternals();
+
+  /**
+   * The step controlled by the label.
+   */
+  public get step(): SbbStepElement | null {
+    return this._step;
+  }
+
+  private _abort = new ConnectedAbortController(this);
+  private _stepper: SbbStepperElement | null = null;
+  private _step: SbbStepElement | null = null;
+
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    const signal = this._abort.signal;
+    this.id = this.id || `sbb-step-label-${nextId++}`;
+    this._stepper = this.closest('sbb-stepper');
```

Comment:
Should we use `closest?.(` for SSR?

Reply:
Idk if I can use the optional chaining like that, anyway the `closest()` method should be null safe.